### PR TITLE
Simplify SSO Integration test with Dex

### PR DESCRIPTION
**Pull request by cniackz**
On Sunday, April 24th 2022 @harshavardhana suggested to simplify our `SSO Integration` test by using https://github.com/minio/minio-iam-testing, I contacted Donatello and learned how Dex works with MinIO. From that learning I came up with this solution, no more Keycloak configuration.

All testing instructions are located in: https://github.com/cniackz/minio/wiki/SSO-Integration-test-simplify

Explanation:

Instead of using keycloak, this time I am using Dex, which is much simpler and faster for testing.
Now there are many things I have changed already, but I will point out the main changes here:

1. Removed big json file to configure keycloak as Dex yaml is very small.
2. Instead of using an ugly bash script to get the code and state, this time I am using a beatiful python soup.
3. It is worth to mention that execution time was reduced significantly.

> Down to only 3m 17s, amazing!

Screen Shot 2022-05-05 at 9 34 09 AM

### CODE REVIEW

**Review by harshavardhana — CHANGES_REQUESTED** [review]
@cniackz you know that you can use our published containers here? 

```
	podman run \
    --name dex \
    --pod iam-testing \
    --detach \
    quay.io/minio/dex:latest
```

and server config is 

```
export MINIO_IDENTITY_OPENID_CONFIG_URL="http://localhost:5556/dex/.well-known/openid-configuration"
export MINIO_IDENTITY_OPENID_CLIENT_ID="minio-client-app"
export MINIO_IDENTITY_OPENID_CLIENT_SECRET="minio-client-app-secret"
# export MINIO_IDENTITY_OPENID_CLAIM_NAME="groups"
export MINIO_IDENTITY_OPENID_SCOPES="openid,groups"
export MINIO_IDENTITY_OPENID_REDIRECT_URI="http://127.0.0.1:10000/oauth_callback"
# role policy if desired:
export MINIO_IDENTITY_OPENID_ROLE_POLICY="consoleAdmin"

export MINIO_ROOT_USER=minio
export MINIO_ROOT_PASSWORD=minio123

# OR
mc admin config set myminio identity_openid \
    config_url="http://localhost:5556/dex/.well-known/openid-configuration" \
    client_id="minio-client-app" \
    client_secret="minio-client-app-secret" \
    scopes="openid,groups" \
    redirect_uri="http://127.0.0.1:10000/oauth_callback" \
    role_policy="consoleAdmin"

# OR
mc admin config set myminio identity_openid:dextest \
    config_url="http://localhost:5556/dex/.well-known/openid-configuration" \
    client_id="minio-client-app" \
    client_secret="minio-client-app-secret" \
    scopes="openid,groups" \
    redirect_uri="http://127.0.0.1:10000/oauth_callback" \
    role_policy="consoleAdmin"



./minio server --console-address ":10000" /tmp/disk
```

**Comment by cniackz** [comment]
Thank @harshavardhana for the review, tomorrow or in a near future I will be addressing this requested change.

**Comment by cniackz** [comment]
Hello @harshavardhana, I have used our Published images and configured server accordingly.
Please kindly review it again and thank you for your time 👍

File: sso-integration/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM ghcr.io/dexidp/dex:latest
+
+ADD config.docker.yaml /etc/dex/
```
Comment (donatello) [inline sso-integration/Dockerfile:3]: Is this file needed anymore since you are using our published image?

**Review by donatello — APPROVED** [review]
Looks good, however is it running in the CI?

**Review by dvaldivia — APPROVED** [review]

File: sso-integration/Dockerfile (hunk 1 above)
Comment (dvaldivia) [reply sso-integration/Dockerfile:3]: it runs on github actions :)

**Comment by cniackz** [comment]
@donatello what do you mean by CI?, continuous integration?. The short answer is yes, the Python code can show that by getting the code and the state that later on we use in our Console API to get the token, if token is obtained, then we pass the test as in our other integration tests 👍

**Pull request merged**